tests/python/gpu/test_nccl.py

```diff
@@ -21,7 +21,16 @@
 
 shapes = [(10), (100), (1000), (10000), (100000), (2,2), (2,3,4,5,6,7,8)]
 keys = [1,2,3,4,5,6,7]
-gpus = range(1,1+len(mx.test_utils.list_gpus()))
+num_gpus = len(mx.test_utils.list_gpus())
+
+#TODO The test is capped to run with 8 GPUs only. This is due to a bug in NCCL 2.1
+#It is addressed in github issue https://github.com/apache/incubator-mxnet/issues/9004
+#Remove this constraint when the github issue is fixed.
+
+if num_gpus > 8 :
+    num_gpus = 8;
+
+gpus = range(1,1+num_gpus)
```

Comment:
Please print a warning into the log so we're actively getting remembered as soon as we switch to new instances.

Reply:
Thanks @marcoabreu  for the review! I have added the print statement. The output of the test is given below.
Regarding machine with 2 gpus, the test caps to 8 gpus only if the machine has more than 8 gpus. The test should pass on machine with 2 gpus

Test output:

gpu]$ python test_nccl.py 
The machine has 16 gpus. We will run the test on 8 gpus only due to a bug in NCCL 2.1.
Please remove this limitation when issue #9004 is fixed.
Passed

Reply:
Great, thanks a lot for the detailed explanation and incorporating the change!